File: Adidas/adidas-fliteTrakr/java/src/main/java/com/nestis/adidas/fliteTrakr/model/Airport.java
package com.nestis.adidas.fliteTrakr.model;

import java.util.ArrayList;
import java.util.List;

/**
 * Class that defines an Airport.
 * @author nestis
 *
 */
public class Airport {

	/**
	 * Airport name;
	 */
	private String origin;

	/**
	 * List of possible destinations. Using a list let us add the same destination twice
	 */
	private List<Destination> destinations;

	public String getOrigin() {
		return origin;
	}
	
	public void setOrigin(String origin) {
		this.origin = origin;
	}

	public void addDestination(Airport destination, Float price) {
		destinations.add(new Destination(destination, price));
	}
	
	public List<Destination> getDestinations() {
		return destinations;
	}

	public Airport() {
		destinations = new ArrayList<Destination>();
	}
	
	public Airport(String origin) {
		destinations = new ArrayList<Destination>();
		this.origin = origin;
	}
	
}

File: Adidas/adidas-fliteTrakr/js/src/model/airport.js
/**
 * Airport definition.
 */
class Airport {

    /**
     * Constructor.
     * @param {*} name City name.
     * @param {*} destinations Destinations. Array of Airport.
     */
    constructor(name, destinations) {
        this.name = name;
        this.destinations = destinations || [];
    }

    /**
     * Adds a new destination.
     * @param {*} dest Airport to add.
     */
    addDestination(dest) {
        this.destinations.push(dest);
    }
}

module.exports = Airport;
